### PR TITLE
feat(TearsheetShell): specify additional floating menu selectors

### PR DESCRIPTION
**Pull request by liunate**
Contributes to #4985 

Allow `<TearsheetShell>` to focus on elements matching CSS selectors specified

#### What did you change?

Pass thru and join `<TearsheetShell selectorsFloatingMenus>` value to the underlying carbon `<ComposedModal selectosFloatingMenus>`

#### How did you test and verify your work?

I add the `selectorsFloatingMenus` property on the one of the story render property in `Tearsheet.stories.jsx` such like:

```jsx
      <div ref={ref}>
        <Tearsheet
          selectorsFloatingMenus={['#helloMan']}
        >
          {mainContent}
        </Tearsheet>
        <div id="helloMan" style={{position: 'fixed', left: '16rem', top: '16rem', zIndex: 9999, backgroundColor: 'green'}}>
            <h1> hello </h1>
            <input type='text' defaultValue="abc" />
          </div>
      </div>
```

Without the `#helloMan` being specified, the opened tearshet does not allow focusing on the input. When the selector is specified, the input can be focused and type inside.

### CODE REVIEW

**Comment by liunate** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by makafsal** [comment]
@liunate Code looks fine to me. But could you please add a storybook for this to demonstrate the feature?

File: packages/ibm-products/src/components/Tearsheet/TearsheetShell.stories.jsx [hunk 1]
```diff
@@ -103,8 +105,8 @@ const Template = ({ influencer, open: _open, slug, ...args }, context) => {
   return (
     <div ref={ref}>
       <Button onClick={() => setOpen(true)}>
-        {beenOpen ? 'Reopen the' : 'Open the'} context.component.componentName
-      </Button>{' '}
+        {`${beenOpen ? 'Reopen' : 'Open'} the ${context.component.displayName}`}
+      </Button>
```
Comment (liunate) [inline packages/ibm-products/src/components/Tearsheet/TearsheetShell.stories.jsx:109]: I think these part could be rewritten in tidier way like this.

Comment (liunate) [reply packages/ibm-products/src/components/Tearsheet/TearsheetShell.stories.jsx:109]: `context.component.componentName` is no longer value, so I replaced it to show the correct component name. In this case here, the `TearsheetShell`

File: packages/ibm-products/src/components/Tearsheet/TearsheetShell.stories.jsx [hunk 2]
```diff
@@ -140,3 +142,72 @@ export const NoAttributesSet = Template.bind({});
...
+          labelText={`This is a floating DOM element with classname "${className}" added outside of the tearsheet DOM element`}
+          placeholder="Focus and enter something"
+        ></TextInput>
+      </div>,
+      document.body
+    )
+  );
+}
+
+export const FocusOnFloatingMenu = ({ open: _open, ...args }, context) => {
```
Comment (liunate) [inline packages/ibm-products/src/components/Tearsheet/TearsheetShell.stories.jsx:174]: Here is the recording showcasing the new property in action.

https://github.com/carbon-design-system/ibm-products/assets/38996291/6730b015-88fb-4f72-becb-12ad06391708

Comment (liunate) [reply packages/ibm-products/src/components/Tearsheet/TearsheetShell.stories.jsx:174]: It works fine in story `docs` as well

**Comment by liunate** [comment]
Not sure why it [failed](https://github.com/carbon-design-system/ibm-products/actions/runs/9184608468/job/25257212646?pr=5092#step:6:129) at the `ci-check:test:c4p` which succeeded in my local env. Any idea?

**Comment by matthewgallo** [comment]
Looks like something is timing out. I just triggered it to run again

**Comment by matthewgallo** [comment]
Hey @liunate, you just have one conflict to resolve in `TearsheetShell.stories.js`.

File: packages/ibm-products/src/components/Tearsheet/TearsheetShell.stories.jsx [hunk 3]
```diff
@@ -179,6 +181,75 @@ NoAttributesSet.args = {
 };
 
 export const ReturnFocusToOpenButton = ReturnFocusTemplate.bind({});
-NoAttributesSet.args = {
+ReturnFocusToOpenButton.args = {
```
Comment (liunate) [inline packages/ibm-products/src/components/Tearsheet/TearsheetShell.stories.jsx:184]: I believe this story template to specify the `.args` was wrong, so updated this to `ReturnFocusToOpenButton`.

**Comment by liunate** [comment]
I just have this PR branch synced again with `main`, see if the changes now looks good for you?

**Review by elycheea — COMMENTED** [review]
Hey Nate! Thank you for taking the time to contribute this! I noticed that we only show the docs and examples within the `TearsheetShell`, but probably should include `selectorsFloatingMenu` in the prop table for `Tearsheet` and `TearsheetNarrow`.

I believe that’s what Afsal was also referring to for including an example. I’ll bring this up in our next team huddle to verify, but I would guess that including the documentation in our Storybook docs[^1] and prop table is higher priority than the specific example for it. :)

[^1]: https://react.carbondesignsystem.com/?path=/docs/components-composedmodal--overview#focus-management

**Comment by liunate** [comment]
Tks 👍 

The reason I didn't add examples to the enhanced components such as `Tearsheet` and `TearsheetNarrow` is because I feel the features that are commonly seen in both components might be better explained in their common base component story, which is the `TearsheetShell`. 

It also avoids duplicate demonstrations that might increase the user's cognitive load and distract them from the unique features of those enhanced components.

**Review by makafsal — DISMISSED** [review]
Looks fine for me

**Review by elycheea — COMMENTED** [review]
@liunate I think let’s get the `selectorsFloatingMenus` prop added to `Tearsheet.tsx` and `TearsheetNarrow.tsx` so they appear in the Storybook props table.

**Comment by liunate** [comment]
@elycheea done!

**Review by elycheea — APPROVED** [review]
謝啦！ LGTM!

**Review by makafsal — APPROVED** [review]
🚀

**Pull request merged**